### PR TITLE
Fixes bug of path directories not working correctly with interactive UI

### PR DESCRIPTION
**Pull request by cdrage**
Fixes the path bug when using `odo create` and specifying either:
  - A path outside of the location of the command being ran
  - ./
  - Blank input

This also changes the order of `odo create`. We ask for the type of
deployment (binary, local or git) FIRST and then the directory.

Closes https://github.com/openshift/odo/issues/1676

### CODE REVIEW

**Comment by surajnarwade** [comment]
> This also changes the order of odo create. We ask for the type of
deployment (binary, local or git) FIRST and then the directory.

does this mean, it is mandatory to provide `--binary` or `--git` flag ?

**Comment by cdrage** [comment]
Ready for review!

@girishramnani @mohammedzee1000 @surajnarwade 

To test:

```sh
git clone https://github.com/openshift/nodejs-ex
cd nodejs-ex
odo create
```

Then go through the options of using "local".

**Comment by girishramnani** [comment]
Would it be possible to have test for this, UT/E2E?

**Comment by cdrage** [comment]
@girishramnani 

Would have to create a new test suite. At the moment there are no integration or unit tests for interactive mode.

File: pkg/odo/cli/component/create.go [hunk 1]
```diff
@@ -270,42 +288,55 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
...
+			co.componentContext = ui.EnterInputTypePath("path", currentDirectory, currentDirectory)
+
+			// Get the correct source location
+			sourceLocation, err := getSourceLocation(co.componentContext, currentDirectory)
+			if err != nil {
+				return errors.Wrapf(err, "unable to get source location")
 			}
+			co.componentSettings.SourceLocation = &sourceLocation
+
+			//glog.V(4).Infof("Selected source path: %s, context %s", *(co.componentSettings.SourceLocation), co.componentContext)
```
Comment (mik-dass) [inline pkg/odo/cli/component/create.go]: Comment found

**Comment by cdrage** [comment]
@mik-dass Updated, can you LGTM / approve?

**Pull request merged**